firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/analytics/AnalyticsConnectorReceiver.java

```diff
@@ -66,6 +67,29 @@ public boolean register() {
     analyticsConnectorHandle =
         analyticsConnector.registerAnalyticsConnectorListener(CRASHLYTICS_ORIGIN, this);
 
+    if (analyticsConnectorHandle == null) {
+      Logger.getLogger()
+          .d(
+              "Could not register AnalyticsConnectorListener with origin "
+                  + "\""
+                  + CRASHLYTICS_ORIGIN
+                  + "\"");
+      // Older versions of FA don't support CRASHLYTICS_ORIGIN. We can try using the old Firebase
+      // Crash Reporting origin
+      analyticsConnectorHandle =
+          analyticsConnector.registerAnalyticsConnectorListener(LEGACY_CRASH_ORIGIN, this);
+
+      // If FA allows us to connect with the legacy origin, but not the new one, nudge customers
+      // to update their FA version.
+      if (analyticsConnectorHandle != null) {
```

Comment:
Would it be worth consolidating this with the `return` value below?

Reply:
I couldn't come up with a way to do that which *helped readability* and the performance impact is so close to nil that it isn't worth optimizing. We could have fewer null checks but we'd need to sprinkle return statements throughout the method. So I (mildly) prefer it as-is.

Reply:
👍